lib/src/test/resources/compileContracts.sh

```diff
@@ -1,47 +1,134 @@
+
+  echo "Copying binaries..."
+  rsync -aW "$REPO_ROOT/cairo$VERSION_SHORT/target/release/" "$(dirname "$0")/$OUT_DIR/cairo/bin/"
+  echo "Copying corelib..."
+  rsync -aW "$REPO_ROOT/cairo$VERSION_SHORT/corelib/" "$(dirname "$0")/$OUT_DIR/cairo/corelib/"
+}
+
+fetch_compilers() {
+  local VERSION=$1
+  local OUT_DIR=$2
+
+  local VERSION_SHORT=${VERSION:0:1}
+
+  echo "Fetching compiler v$VERSION binaries..."
+
+  mkdir -p "$OUT_DIR"
+
+  OS=$(uname)
+  ARCH=$(uname -m)
+
+  if [ "$OS" == "Linux" -a "$ARCH" == "x86_64" ]; then
+    curl -LsSf "https://github.com/starkware-libs/cairo/releases/download/v${VERSION}/release-x86_64-unknown-linux-musl.tar.gz" | tar -xz -C "$OUT_DIR" || exit 1
+  elif [ "$OS" == "Darwin" -a "$ARCH" == "arm64" ]; then
+    curl -LsSf "https://github.com/starkware-libs/cairo/releases/download/v${VERSION}/release-aarch64-apple-darwin.tar" | tar -x -C "$OUT_DIR" || exit 1
+  elif [ "$OS" == "Darwin" -a "$ARCH" == "x86_64" ]; then
+    build_cairo_compilers "$VERSION" "$OUT_DIR"
+  else
+    echo "Unsupported OS or architecture: $ARCH-$OS"
+    exit 1
+  fi
+
+  COMPILER_PATH=$(pwd)/"$OUT_DIR"/cairo/bin/starknet-compile
+  SIERRA_PATH=$(pwd)/"$OUT_DIR"/cairo/bin/starknet-sierra-compile
+
+  echo "Done!"
+}
+
+pushd "$(dirname "$0")" || exit 1
+mkdir -p "$V0_ARTIFACT_PATH"
 
 echo "Compiling v0 contracts.."
 
 while IFS= read -r -d '' file; do
   name="$(basename -- "$file" .cairo)"
   if [[ $name == *"account"* ]]; then
-    starknet-compile-deprecated "$file" --account_contract --output "compiled_v0/$name.json" --abi "compiled_v0/${name}Abi.json"
+    starknet-compile-deprecated "$file" --account_contract --output "$V0_ARTIFACT_PATH/$name.json" --abi "$V0_ARTIFACT_PATH/${name}Abi.json" || exit 1
   else
-    starknet-compile-deprecated "$file" --output "compiled_v0/$name.json" --abi "compiled_v0/${name}Abi.json"
+    starknet-compile-deprecated "$file" --output "$V0_ARTIFACT_PATH/$name.json" --abi "$V0_ARTIFACT_PATH/${name}Abi.json" || exit 1
   fi
-done < <(find src_v0 -name "*.cairo" -type f -print0)
+done < <(find "$V0_CONTRACT_PATH" -name "*.cairo" -type f -print0)
 popd
 echo "Done!"
 
-if ! which cargo > /dev/null; then
-  echo "Installing rust..."
-  curl https://sh.rustup.rs -sSf | sh -s -- -y
-fi
+pushd "$(dirname "$0")" || exit 1
+fetch_compilers "$V1_COMPILER_VERSION" "$V1_COMPILER_BIN_PATH"
+mkdir -p "$V1_ARTIFACT_PATH"
 
-if ! find ../cairo/target/debug -name starknet-compile -type f > /dev/null || ! find ../cairo/target/debug -name starknet-sierra-compile -type f > /dev/null; then
-  echo "Building starknet compiler"
-  pushd ../cairo || exit 1
-  cargo build
-  cargo run --bin starknet-compile -- --version
-  cargo run --bin starknet-sierra-compile -- --version
-  popd
-fi
+echo "Compiling v1 contracts.."
 
-COMPILER_PATH=$(pwd)/../cairo/target/debug/starknet-compile
-SIERRA_PATH=$(pwd)/../cairo/target/debug/starknet-sierra-compile
+while IFS= read -r -d '' file; do
```

Comment:
it seems v1 and v2 are pretty similar, maybe we could do this in for loop?

Reply:
They have different flags, I think making it a loop and specifying what flags go to which loop would make it less readable.

Reply:
we could loop over specific compile commands, seems to me this is the only part that varies, right?